src/selectors/disabled.js

```diff
@@ -84,28 +144,100 @@ export const getNodeDisabled = createSelector(
     nodeIDs,
     nodeDisabledNode,
     nodeDisabledTag,
-    nodeDisabledModularPipeline,
+    nodeDisabledModularPipelineFilter,
     nodeDisabledPipeline,
     nodeType,
     typeDisabled
   ) =>
-    arrayToObject(nodeIDs, (id) =>
-      [
-        nodeDisabledNode[id],
-        nodeDisabledTag[id],
-        nodeDisabledModularPipeline[id],
-        nodeDisabledPipeline[id],
-        typeDisabled[nodeType[id]],
-      ].some(Boolean)
+    arrayToObject(
+      nodeIDs,
+      (id) =>
+        nodeDisabledNode[id] ||
+        nodeDisabledTag[id] ||
+        nodeDisabledModularPipelineFilter[id] ||
+        nodeDisabledPipeline[id] ||
+        typeDisabled[nodeType[id]]
+    )
+);
+
+/**
+ * Set disabled status if the modular pipeline's children are all disabled
+ */
+export const getModularPipelineDisabledChildren = createSelector(
+  [
+    getModularPipelineIDs,
+    getNodeDisabledNode,
+    getNodeDisabledTag,
+    getNodeDisabledModularPipelineFilter,
+    getNodeDisabledPipeline,
+    getNodeType,
+    getNodeTypeDisabled,
+    getModularPipelineChildren,
+  ],
+  (
+    modularPipelineIDs,
+    nodeDisabledNode,
+    nodeDisabledTag,
+    nodeDisabledModularPipelineFilter,
+    nodeDisabledPipeline,
+    nodeType,
+    typeDisabled,
+    modularPipelineChildren
+  ) =>
+    arrayToObject(modularPipelineIDs, (modPipID) => {
+      const children = Object.keys(modularPipelineChildren[modPipID]);
+      const allChildrenAreDisabled = children.every(
+        (nodeID) =>
```

Comment:
Is this function the same as used in `getNodeDisabledExclModPip`? Might be a good idea to move this out to a function if so.

Reply:
ah good catch! I think I added `getNodeDisabledExclModPip` after adding this selector. I can just import the result of that selector instead, which will simplify this one tremendously.